test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
@stevekuznetsov You are talking about updating to this comment: Waiting: Determining build queue position

?

I think because it doesn't need to be in this state to move on from any waiting state.